onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlDFT.h

```diff
@@ -67,46 +67,55 @@ class GpuDFTOperator : public WRL::Base<IMLOperatorKernel>
     ComPtr<ID3D12RootSignature> m_rootSignature;
     ComPtr<ID3D12PipelineState> m_pipelineState;
 
-    std::vector<uint32_t> m_inputDims = {};
-    std::vector<uint32_t> m_outputDims = {};
     int64_t m_axis;
     bool m_isOnesided;
     bool m_isInverse;
 
-    uint32_t m_outputDataSize = 0;
-    uint32_t m_inputDataSize = 0;
-    uint32_t m_outputIdx = 0;
-    uint32_t m_numPasses = 0;
-
-    // Allocate temporary buffers if needed
-    struct ResourceDesc
-    {
-        ComPtr<ID3D12Resource> Resource;
-        std::array<uint32_t, 4> Sizes;
-        std::array<uint32_t, 4> Strides;
-    };
-    std::vector<ResourceDesc> m_resourceLoopList = {};
+    struct StockhamParameters {
+        // Allocate temporary buffers if needed
+        struct ResourceDesc
+        {
+            ComPtr<ID3D12Resource> Resource;
+            std::array<uint32_t, 4> Sizes;
+            std::array<uint32_t, 4> Strides;
+        };
```

Comment:
can the "4" have a name or an explanatory comment?

Reply:
If applicable, we have `NchwDimensionCount` already. Though, I'm not sure the semantics of DFT (if they are not specifically batch, channel, height, width, then we should not use that, as it would be more confusing).

Reply:
The semantics of the DFT are:
// { before_dft_axis, axis, after_dft_axis, real_or_complex }

There is a comment where the DFT reshape takes place.